crates/rolldown/src/utils/chunk/deconflict_chunk_symbols.rs

```diff
@@ -99,9 +100,18 @@ pub fn deconflict_chunk_symbols(
   if matches!(format, OutputFormat::Esm) {
     chunk.imports_from_external_modules.iter().for_each(|(module, _)| {
       let db = link_output.symbol_db.local_db(*module);
-      db.classic_data.iter_enumerated().skip(1).for_each(|(symbol, _)| {
-        renamer.add_symbol_in_root_scope((*module, symbol).into());
+      db.classic_data.iter_enumerated().for_each(|(symbol, _)| {
+        let symbol_ref = (*module, symbol).into();
+        if link_output.used_symbol_refs.contains(&symbol_ref) {
+          renamer.add_symbol_in_root_scope(symbol_ref);
+        }
       });
+      for symbol_id in db.ast_scopes.facade_symbol_classic_data().keys() {
```

Comment:
I changed to filter by `used_symbol_refs` because we separate `facade_scoping` and `scoping`, we can't use that trick anymore.